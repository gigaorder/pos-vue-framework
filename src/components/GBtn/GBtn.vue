<template>
	<button v-ripple :class="classes" :style="styles" @click="onClick">
		<span class="g-btn__content">
			<slot></slot>
		</span>
	</button>
</template>

<script>
  import { computed } from '@vue/composition-api';
  import { convertToGradient, convertToUnit } from '../../utils/helpers';
  import Ripple from '../../directives/ripple/ripple';

  export default {
    name: 'GBtn',
    directives: {
      Ripple
    },
    props: {
      //classes
      ...{
        raised: Boolean,
        depressed: Boolean,
        disabled: Boolean,
        rounded: Boolean,
        text: Boolean,
        fab: Boolean,
        tile: Boolean,
        icon: Boolean,
        outlined: Boolean,
        dashed: Boolean,
        active: Boolean,
        activeClass: {
          type: String,
          default: 'g-btn__active',
        }
      },
      //style
      ...{
        elevation: [String, Number],
        gradientAngle: { type: String, default: '45deg' },
        absolute: Boolean,
        fixed: Boolean,
        top: Boolean,
        bottom: Boolean,
        left: Boolean,
        right: Boolean,
        large: Boolean,
        small: Boolean,
        xLarge: Boolean,
        xSmall: Boolean,
        maxHeight: [String, Number],
        maxWidth: [String, Number],
        minHeight: [String, Number],
        minWidth: [String, Number],
        block: Boolean,
        width: [String, Number],
        height: [String, Number],
        textColor: String,
        color: String,
        backgroundColor: String,
        gradient: String,
      }
    },
    setup(props, context) {
      let classes = computed(() => {
        let _classes = {
          'g-btn': true,
          'g-btn__raised': props.raised,
          'g-btn__flat': isFlat.value,
          'g-btn__tile': props.tile,
          'g-btn__fab': props.fab,
          'g-btn__icon': props.icon,
          'g-btn__text': props.text,
          'g-btn__top': props.top,
          'g-btn__fixed': props.fixed,
          'g-btn__absolute': props.absolute,
          'g-btn__bottom': props.bottom,
          'g-btn__left': props.left,
          'g-btn__right': props.right,
          'g-btn__block': props.block,
          'g-btn__depressed': props.depressed || props.outlined,
          'g-btn__disabled': props.disabled,
          'g-btn__rounded': props.rounded,
          'g-btn__outlined': props.outlined,
          'g-btn__round': isRound.value,
          'g-btn__contained': contained.value,
          [props.activeClass]: props.active,
          ...elevationClasses.value
        };

        let size = '';

        if (props.large) {
          size = 'g-size__large';
        } else if (props.small) {
          size = 'g-size__small';
        } else if (props.xSmall) {
          size = 'g-size__x-small';
        } else if (props.xLarge) {
          size = 'g-size__x-large';
        } else {
          size = 'g-size__default';
        }
        if (size) {
          _classes[size] = true;
        }

        if (props.gradient) {
          if (props.gradient.toString().includes('-')) {
            _classes[props.gradient] = true;
          }
        }

        let elevationClassName = props.elevation ? `g-btn__elevation-${props.elevation}` : null;
        if (elevationClassName) {
          _classes[elevationClassName] = true;
        }

        return _classes;
      });

      let elevationClasses = computed(() => {
        const elevation = props.elevation;
        if (!!elevation) {
          return {};
        }

        if (isNaN(parseInt(elevation))) {
          return {};
        }

        return { [`g-btn__elevation-${props.elevation}`]: true }
      });

      let isRound = computed(() => {
        return props.icon || props.fab;
      });

      let isFlat = computed(() => {
        return props.text || props.icon || props.outlined;
      });

      let contained = computed(() => {
        return !isFlat.value && !props.depressed && !props.elevation;
      });

      let styles = computed(() => {
        let _styles = {
          ...props.textColor && { color: props.textColor.replace('-', '') },
          ...props.backgroundColor && { backgroundColor: props.backgroundColor.replace('-', '') },
          ...props.width && { width: convertToUnit(props.width) },
          ...props.height && { height: convertToUnit(props.height) },
          ...props.maxWidth && { maxWidth: convertToUnit(props.maxWidth) },
          ...props.maxHeight && { maxHeight: convertToUnit(props.maxHeight) },
          ...props.minWidth && { minWidth: convertToUnit(props.minWidth) },
          ...props.minHeight && { minHeight: convertToUnit(props.minHeight) },
        };

        // Params: linear-gradient(45deg, yellow, green)
        if (props.gradient && !props.gradient.toString().includes('-')) {
          _styles['background-image'] = convertToGradient(props.gradient.toString().split(','), props.gradientAngle);
        }

        return _styles;
      });

      let onClick = (event) => {
        context.emit('click', event);
      };

      return {
        classes,
        styles,
        onClick,
      }
    }
  }
</script>

<style scoped>

</style>
