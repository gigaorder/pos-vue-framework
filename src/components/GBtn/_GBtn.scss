button {
  outline: none;
  font-family: $font-family-base;
  font-weight: $font-weight-base;
  cursor: pointer;
  color: rgba(0, 0, 0, .87);
  -webkit-tap-highlight-color: transparent;
}

//Vuetify
.g-btn {
  align-items: center;
  border: none;
  border-radius: 4px;
  display: inline-flex;
  flex: 0 0 auto;
  font-weight: $btn-font-weight;
  letter-spacing: $btn-letter-spacing;
  justify-content: center;
  max-width: 100%;
  outline: 0;
  position: relative;
  text-decoration: none;
  text-indent: $btn-letter-spacing;
  text-transform: uppercase;
  transition-duration: $btn-transition-duration;
  transition-property: box-shadow, transform, opacity;
  transition-timing-function: $btn-transition-fn;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;

  &:hover::before {
    opacity: 0.04;
  }

  &:focus::before {
    opacity: 0.12;
  }

  &__active {
    &:hover::before,
    &::before {
      opacity: 0.12;
    }

    &:focus::before {
      opacity: 0.16;
    }
  }

  color: rgba(0, 0, 0, 0.87);

  &.g-btn__disabled {
    color: rgba(0, 0, 0, 0.26) !important;

    .g-icon {
      color: rgba(0, 0, 0, 0.26) !important;
    }

    &:not(.g-btn__flat):not(.g-btn__text):not(.g-btn__outlined) {
      background-color: rgba(0, 0, 0, 0.12) !important;
    }
  }

  &:not(.g-btn__flat):not(.g-btn__text):not(.g-btn__outlined) {
    background-color: #f5f5f5;
  }

  &.g-btn__outlined.g-btn__text {
    border-color: rgba(0, 0, 0, 0.12);
  }

  &.g-btn__icon {
    color: rgba(0, 0, 0, 0.54);
  }

  @each $name, $size in $btn-font-sizes {
    &.g-size__#{$name} {
      font-size: $size;
    }
  }

  &:before {
    border-radius: inherit;
    bottom: 0;
    color: inherit;
    content: '';
    left: 0;
    opacity: 0;
    pointer-events: none;
    position: absolute;
    right: 0;
    top: 0;
    transition: $btn-transition
  }
;

  &:before {
    background-color: currentColor;
  }

  &:not(.g-btn__disabled) {
    will-change: box-shadow;
  }

  &:not(.g-btn__round):not(.g-btn__fab):not(.g-btn__icon) {
    @each $name, $size in $btn-sizes {
      &.g-size__#{$name} {
        height: #{$size}px;
        min-width: #{round($size * 1.777777777777778)}px; // default ratio;
        padding: 0 #{$size / 2.25}px;
      }
    }
  }

  > .g-btn__content .g-icon {
    color: inherit;
  }
}

.g-btn__content {
  align-items: center;
  color: inherit;
  display: flex;
  flex: 1 0 auto;
  justify-content: inherit;
  line-height: normal;
  position: relative;

  .g-icon__left,
  .g-icon__right {
    font-size: $btn-icon-font-size;
    height: $btn-icon-font-size;
    width: $btn-icon-font-size;
  }

  .g-icon__left {
    margin-left: -4px;
    margin-right: 8px;
  }

  .g-icon__right {
    margin-left: 8px;
    margin-right: -4px;
  }
}

//Modifier
// Modifiers
.g-btn:not(.g-btn__text):not(.g-btn__outlined) {
  &.g-btn__active:before {
    opacity: $btn-active-opacity;
  }

  &:hover:before {
    opacity: $btn-hover-opacity;
  }

  &:focus:before {
    opacity: $btn-focus-opacity;
  }
}


.g-btn__absolute,
.g-btn__fixed {
  position: absolute;

  &.g-btn__right {
    right: map-get($grid-gutters, 'sm');
  }

  &.g-btn__left {
    left: map-get($grid-gutters, 'sm');
  }

  &.g-btn__top {
    top: map-get($grid-gutters, 'sm');
  }

  &.g-btn__bottom {
    bottom: map-get($grid-gutters, 'sm');
  }
}

.g-btn__block {
  display: flex;
  flex: 1 0 auto;
  min-width: 100% !important;
}


.g-btn__contained {
  @include elevation(2);

  &:after {
    @include elevation(4);
  }

  &:active {
    @include elevation(8);
  }
}

.g-btn__depressed {
  box-shadow: none !important;
}

.g-btn__disabled {
  box-shadow: none !important;
  pointer-events: none !important;
}

.g-btn__icon,
.g-btn__fab {
  min-height: 0;
  min-width: 0;
  padding: 0;

  @each $name, $size in $fab-icon-sizes {
    &.g-size__#{$name} {
      .g-icon {
        height: #{$size}px !important;
        font-size: #{$size}px !important;
        width: #{$size}px !important;
      }
    }
  }
}

.g-btn__icon {
  @each $name, $size in $btn-sizes {
    &.g-size__#{$name} {
      height: #{$size}px !important;
      width: #{$size}px !important;
    }
  }
}

.g-btn__fab {
  &.g-btn__contained {
    @include elevation(6);

    &:after {
      @include elevation(8);
    }

    &:active {
      @include elevation(12)
    }
  }

  &.g-btn__fixed,
  &.g-btn__absolute {
    z-index: 4;
  }

  @each $name, $size in $fab-sizes {
    &.g-size__#{$name} {
      height: #{$size}px;
      width: #{$size}px;
    }

    &.g-btn__absolute {
      &.g-btn__bottom {
        bottom: -#{$size / 2}px;
      }

      &.g-btn__top {
        top: -#{$size / 2}px;
      }
    }
  }
}

.g-btn__fixed {
  position: fixed;
}

.g-btn__outlined {
  border: $btn-outline-border-width solid currentColor;

  &:before {
    border-radius: 0
  }
}

.g-btn__outlined,
.g-btn__round {
  .g-btn__content {
    .g-icon {
      color: currentColor;
    }
  }
}

.g-btn__outlined,
.g-btn__flat,
.g-btn__text {
  background-color: transparent;
}

.g-btn__round,
.g-btn__rounded {
  &:before {
    border-radius: inherit;
  }
}

.g-btn__round {
  border-radius: 50%
}

.g-btn__rounded {
  border-radius: $btn-rounded-border-radius
}

.g-btn__tile {
  border-radius: 0
}

$elevationMin: 1;
$elevationMax: 24;
@for $i from $elevationMin through $elevationMax {
  .g-btn__elevation-#{$i} {
    @include elevation($i);
  }
}

