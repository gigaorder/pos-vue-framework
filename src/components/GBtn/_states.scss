@mixin states($material) {
  &:hover::before {
    opacity: 0.04;
  }

  &:focus::before {
    opacity: 0.12;
  }
}

&--active {
  @include active-states($material);
}

@mixin active-states($material) {
  &:hover::before,
  &::before {
    opacity: 0.12;
  }

  &:focus::before {
    opacity: 0.16;
  }
}
