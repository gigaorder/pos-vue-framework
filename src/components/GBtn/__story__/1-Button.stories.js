import { text, withKnobs, boolean, number, array } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import GBtn from '../GBtn';

export default {
  title: 'Button',
  decorators: [withKnobs],
};

export const test1 = () => ({
  components: { GBtn },
  props: {
    text: { default: text('Button Text', 'XLARGE BUTTON') },
    outlined: { type: Boolean, default: boolean('outlined', true) }
  },
  template: `<g-btn :outlined="outlined" raised elevation='0' x-large>{{text}}</g-btn>`,
})

export const GBtnDefault = () => ({
  components: { GBtn },
  props: {
    text: { default: text('Button Text', 'NORMAL BUTTON') },
    flat: { type: Boolean, default: boolean('flat', false) },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    elevation: { type: Number, default: number('elevation', 0) },
    block: { type: Boolean, default: boolean('block', false) },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    tile: { type: Boolean, default: boolean('tile', false) },
  },
  template: `<g-btn :tile="tile" :flat="flat" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled" :block="block" :elevation="elevation" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">{{text}}</g-btn>`,
})

export const GBtnOutlined = () => ({
  components: { GBtn },
  props: {
    text: { default: text('Button Text', 'OUTLINED BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    outlined: { type: Boolean, default: boolean('outlined', true) },
    textColor: { default: text('textColor', 'blue') },
    block: { type: Boolean, default: boolean('block', false) },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    tile: { type: Boolean, default: boolean('tile', false) },

  },
  template: `<g-btn :tile="tile" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled"  :block="block" :text-color="textColor" :outlined="outlined" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">{{text}}</g-btn>`,
})

export const GBtnDepressed = () => ({
  components: { GBtn },
  props: {
    text: { default: text('Button Text', 'DEPRESSED BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    depressed: { type: Boolean, default: boolean('depressed', true) },
    textColor: { default: text('textColor', 'orange') },
    block: { type: Boolean, default: boolean('block', false) },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    tile: { type: Boolean, default: boolean('tile', false) },

  },
  template: `<g-btn :tile="tile" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled"  :block="block" :text-color="textColor" :depressed="depressed" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">{{text}}</g-btn>`,
})

export const GBtnText = () => ({
  components: { GBtn },
  props: {
    textValue: { default: text('Button Text', 'TEXT BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    text: { type: Boolean, default: boolean('text', true) },
    textColor: { default: text('textColor', 'green') },
    block: { type: Boolean, default: boolean('block', false) },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    tile: { type: Boolean, default: boolean('tile', false) },

  },
  template: `<g-btn :tile="tile" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled" :block="block" :text="text" :text-color="textColor" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">{{textValue}}</g-btn>`,
})


export const GBtnRounded = () => ({
  components: { GBtn },
  props: {
    textValue: { default: text('Button Text', 'ROUNDED BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    rounded: { type: Boolean, default: boolean('rounded', true) },
    textColor: { default: text('textColor', 'white') },
    backgroundColor: { default: text('backgroundColor', 'brown') },
    block: { type: Boolean, default: boolean('block', false) },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    tile: { type: Boolean, default: boolean('tile', false) },

  },
  template: `<g-btn :tile="tile" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled"  :block="block" :background-color="backgroundColor" :rounded="rounded" :text-color="textColor" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">{{textValue}}</g-btn>`,
})


export const GBtnFab = () => ({
  components: { GBtn },
  props: {
    textValue: { default: text('Button Text', 'ROUNDED BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    fab: { type: Boolean, default: boolean('fab', true) },
    textColor: { default: text('textColor', 'white') },
    backgroundColor: { default: text('backgroundColor', 'blue') },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', true) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', true) },
    tile: { type: Boolean, default: boolean('tile', false) },

  },
  template: `<g-btn :tile="tile" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :disabled="disabled" :fab="fab" :background-color="backgroundColor" :text-color="textColor" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">
              <i class="material-icons g-icon">android</i>
            </g-btn>`,
})

export const GBtnIcon = () => ({
  components: { GBtn },
  props: {
    textValue: { default: text('Button Text', 'ROUNDED BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    icon: { type: Boolean, default: boolean('icon', true) },
    textColor: { default: text('textColor', 'red') },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },

  },
  template: `<g-btn :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right" :icon="icon" :disabled="disabled" :text-color="textColor" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">
              <i class="material-icons g-icon">star</i>
            </g-btn>`,
})

export const GBtnGradient = () => ({
  components: { GBtn },
  props: {
    textValue: { default: text('Button Text', 'GRADIENT BUTTON') },
    xSmall: { type: Boolean, default: boolean('xSmall', false) },
    small: { type: Boolean, default: boolean('small', false) },
    large: { type: Boolean, default: boolean('large', false) },
    xLarge: { type: Boolean, default: boolean('xLarge', false) },
    textColor: { default: text('textColor', 'white') },
    disabled: { type: Boolean, default: boolean('disabled', false) },
    fixed: { type: Boolean, default: boolean('fixed', false) },
    absolute: { type: Boolean, default: boolean('absolute', false) },
    top: { type: Boolean, default: boolean('top', false) },
    bottom: { type: Boolean, default: boolean('bottom', false) },
    left: { type: Boolean, default: boolean('left', false) },
    right: { type: Boolean, default: boolean('right', false) },
    gradient: { type: String, default: text('gradient', 'yellow, green, blue') },
    gradientAngle: { type: String, default: text('gradientAngle', '120deg') },
    tile: { type: Boolean, default: boolean('tile', false) },
  },
  template: `<g-btn :tile="tile" :gradient-angle="gradientAngle" :gradient="gradient" :absolute="absolute" :fixed="fixed" :top="top" :bottom="bottom" :left="left" :right="right"  :disabled="disabled" :text-color="textColor" :x-small="xSmall" :small="small" :large="large" :x-large="xLarge">
            {{textValue}}
            </g-btn>`,
})

export const test2 = () => ({
  components: { GBtn },
  setup() {
    return () =>
      <g-btn outlined raised elevation='0' x-large vOn:click={action('test123')}>
        {text('Button Text', 'XLARGE BUTTON')}
      </g-btn>
  },
})

import Vue from 'vue/dist/vue.common.js'

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnDefault()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnDepressed()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnFab()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnGradient()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnIcon()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnOutlined()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnRounded()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})


describe('test', function () {
  it('should test1', function () {
    const vm = new Vue(GBtnText()).$mount();
    expect(vm.$el.outerHTML).toMatchSnapshot()
  });
})

