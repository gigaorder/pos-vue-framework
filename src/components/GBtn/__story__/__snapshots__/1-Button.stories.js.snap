// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should test1 1`] = `
<button class="g-btn waves-effect g-btn__raised g-btn__outlined g-size__x-large g-btn__elevation-0"
        style="border: thin solid currentcolor; background-color: transparent;"
>
  XLARGE BUTTON
</button>
`;

exports[`test should test1 2`] = `
<button class="g-btn g-btn__depressed g-size__default waves-effect"
        style="color: rgb(255, 165, 0); caret-color: #FFA500;"
>
  <span class="g-btn__content">
    DEPRESSED BUTTON
  </span>
</button>
`;

exports[`test should test1 3`] = `
<button class="g-btn g-btn__fab g-btn__fixed g-btn__bottom g-btn__right g-btn__round g-btn__contained bg-blue g-size__default waves-effect"
        style="color: rgb(255, 255, 255); caret-color: #ffffff;"
>
  <span class="g-btn__content">
    <i class="material-icons g-icon">
      android
    </i>
  </span>
</button>
`;

exports[`test should test1 4`] = `
<button class="g-btn g-btn__contained g-size__default waves-effect"
        style="color: rgb(255, 255, 255); caret-color: #ffffff;"
>
  <span class="g-btn__content">
    GRADIENT BUTTON
  </span>
</button>
`;

exports[`test should test1 5`] = `
<button class="g-btn g-btn__flat g-btn__icon g-btn__round g-size__default waves-effect"
        style="color: rgb(255, 76, 76); caret-color: #FF4C4C;"
>
  <span class="g-btn__content">
    <i class="material-icons g-icon">
      star
    </i>
  </span>
</button>
`;

exports[`test should test1 6`] = `
<button class="g-btn g-btn__flat g-btn__depressed g-btn__outlined g-size__default waves-effect"
        style="color: rgb(25, 25, 255); caret-color: #1919FF;"
>
  <span class="g-btn__content">
    OUTLINED BUTTON
  </span>
</button>
`;

exports[`test should test1 7`] = `
<button class="g-btn g-btn__rounded g-btn__contained bg-brown g-size__default waves-effect"
        style="color: rgb(255, 255, 255); caret-color: #ffffff;"
>
  <span class="g-btn__content">
    ROUNDED BUTTON
  </span>
</button>
`;

exports[`test should test1 8`] = `
<button class="g-btn g-btn__flat g-btn__text g-size__default waves-effect"
        style="color: rgb(50, 153, 50); caret-color: #329932;"
>
  <span class="g-btn__content">
    TEXT BUTTON
  </span>
</button>
`;
